src/three-components/Debugger.ts

```diff
@@ -0,0 +1,46 @@
+import {Mesh, OrthographicCamera, PlaneBufferGeometry, ShaderMaterial, Texture, WebGLRenderTarget} from 'three';
+
+import {Renderer} from './Renderer';
+
+export interface ModelViewerRendererDebugDetails {
+  renderer: Renderer;
+  THREE: {
+    ShaderMaterial: Constructor<ShaderMaterial>;
+    PlaneBufferGeometry: Constructor<PlaneBufferGeometry>;
+    OrthographicCamera: Constructor<OrthographicCamera>;
+    WebGLRenderTarget: Constructor<WebGLRenderTarget>;
+    Texture: Constructor<Texture>;
+    Mesh: Constructor<Mesh>;
+  };
+}
+
+/**
+ * This Debugger exposes internal details of the <model-viewer> rendering
+ * substructure so that external tools can more easily inspect and operate on
+ * them.
+ *
+ * It also activates shader debugging on the associated GL context. Shader
+ * debugging trades performance for useful error information, so it is not
+ * recommended to activate this unless needed.
+ */
+export class Debugger {
+  constructor(renderer: Renderer) {
+    renderer.renderer.debug = {checkShaderErrors: false};
+    Promise.resolve().then(() => {
+      self.dispatchEvent(new CustomEvent<ModelViewerRendererDebugDetails>(
+          'model-viewer-renderer-debug', {
+            detail: {
+              renderer,
+              THREE: {
+                ShaderMaterial,
+                Texture,
+                Mesh,
+                PlaneBufferGeometry,
+                OrthographicCamera,
+                WebGLRenderTarget
```

Comment:
What's the rationale behind these particular things to include?

Reply:
These are intended to be the minimum viable set of things needed to do debug rendering outside of the element. The actual set we should use may be different, have any suggestions?

Reply:
So basically what's needed to display a texture? Yeah that sounds good. I'm not sure what other kinds of debugging we'd do here.

Reply:
We can add more as we need it